### ISSUE TITLE
Indeca CocaCola Wireless Pro Controller

### ISSUE THREAD

**Issue by ImZealot** [Davidobot/BetterJoyForCemu#101]
Hello, this is the 1st time I report an issue on GitHub, so I'm sorry in advance if I'm doing something wrong.

I recently bought a [3rd-party Pro Controller](https://img.pccomponentes.com/articles/16/169791/2.jpg), by Indeca, and everything works fine on thw Nintendo Switch, but I'm having lots of trouble making it work on the computer. 

Connecting it by cable just charges it, I've tried holding the home button, single press, nothing works.

Connecting it by bluetooth with the dongle I use, but with no software at all, just "default" drivers, made it almost perfectly work. The DPad was non-existing and the Left Analog was the DPad (according to Windows calibration).

I found out about this wonderful software! So I installed it and tried my controller. It's even worse now.

The Left Analog just doesn't work, the Right Analog is associanted with X and Y Rotation, and isn't centered (I think the resting position is all left and all down), the LZ and RZ buttons on my controller look analog, but they're not, and they're being recognised as analog in Z Axis (I don't know if they'll work as buttons). All the other buttons except the capture button work as intended I think. 

[Here ](https://gfycat.com/WiltedDenseCorydorascatfish)is a video of me testing all buttons and showing the issues. (sorry for the quality)

I don't know if it's possible to make this controller work as intended on computer, but maybe this software can make it work if it's coded for it!

Thank you.

**Comment by Davidobot** [comment]
Hi!

The X/Y rotation is how xinput works, as is the mapping of the ZL/ZR onto the z-axis. The pro controller does not have analogue shoulder buttons, so it is meant to snap to values.

The centering thing is a problem indeed, as is the lack of left analogue. Is the controller being recognised as a pro controller by BetterJoyForCemu?

Try calibrating the controller on a Switch and see if that helps. 

Also apologies if this won't lead to anything - I can't provide support for all 3rd party controllers, but I will happily accept pull requests that implement support for them.

**Comment by ImZealot** [comment]
Hello! Thanks a lot for the answer.

Yes, it is recognized as a pro controller by BetterJoyForCemu.

I calibrated both Analog Sticks and the Motion Controls on the Switch and nothing changed...

Ok, so the only issues here are the Left Analog missing and the Right one not being centered, correct?

What would have to be done to correctly configure BetterJoyForCemu for this controller? I notice it reads "Using factory stick calibration data.", if one could change this data, or really just "code" a completely different controller, I'd be happy to learn how to. 

I've got some programming knowledge, not too advanced, but I'm taking a 3-year Managing and Programming Informatic Systems Course.

Thank you for the help.
